process_data.sh

```diff
@@ -146,162 +108,77 @@ cd ${SUBJECT}/anat/
-  sct_extract_metric -i mtsat.nii.gz -f label_axT1w/atlas -l 51 -vert 2:5 -vertfile label_axT1w/template/PAM50_levels.nii.gz -o ${PATH_RESULTS}/MTsat.csv -append 1
-  sct_extract_metric -i t1map.nii.gz -f label_axT1w/atlas -l 51 -vert 2:5 -vertfile label_axT1w/template/PAM50_levels.nii.gz -o ${PATH_RESULTS}/T1.csv -append 1
+  segment_if_does_not_exist $file_t1w "t1" # Do we keep it or not?
+  segment_if_does_not_exist $file_mton "t2s" # or T2
+  #segment_if_does_not_exist $file_mtoff "t1" # or DWI or t1 = NO --> automatic segmentation doesn't work with sct_deepseg_sc
+ 
 else
   echo "WARNING: MTS dataset is incomplete."
 fi
 
+
 # T2s
 # ------------------------------------------------------------------------------
 file_t2s="${SUBJECT}_T2star"
-# Compute root-mean square across 4th dimension (if it exists), corresponding to all echoes in Philips scans.
-sct_maths -i ${file_t2s}.nii.gz -rms t -o ${file_t2s}_rms.nii.gz
-file_t2s="${file_t2s}_rms"
-# Bring vertebral level into T2s space
-sct_register_multimodal -i label_T1w/template/PAM50_levels.nii.gz -d ${file_t2s}.nii.gz -o PAM50_levels2${file_t2s}.nii.gz -identity 1 -x nn
+
+# Segment spinal cord (only if it does not exist)
+segment_if_does_not_exist $file_t2s "t2s"
+
 # Segment gray matter (only if it does not exist)
 segment_gm_if_does_not_exist $file_t2s "t2s"
-file_t2s_seg=$FILESEG
-# Compute the gray matter CSA between C3 and C4 levels
-# NB: Here we set -no-angle 1 because we do not want angle correction: it is too
-# unstable with GM seg, and t2s data were acquired orthogonal to the cord anyways.
-sct_process_segmentation -i ${file_t2s_seg}.nii.gz -angle-corr 0 -vert 3:4 -vertfile PAM50_levels2${file_t2s}.nii.gz -o ${PATH_RESULTS}/csa-GM_T2s.csv -append 1
+
 
 # DWI
 # ------------------------------------------------------------------------------
 file_dwi="${SUBJECT}_dwi"
 cd ../dwi
-# If there is an additional b=0 scan, add it to the main DWI data
-concatenate_b0_and_dwi "${SUBJECT}_acq-b0_dwi" $file_dwi
-file_dwi=$FILE_DWI
-file_bval=${file_dwi}.bval
-file_bvec=${file_dwi}.bvec
-# Separate b=0 and DW images
-sct_dmri_separate_b0_and_dwi -i ${file_dwi}.nii.gz -bvec ${file_bvec}
-# Get centerline
-sct_get_centerline -i ${file_dwi}_dwi_mean.nii.gz -c dwi -qc ${PATH_QC} -qc-subject ${SUBJECT}
-# Create mask to help motion correction and for faster processing
-sct_create_mask -i ${file_dwi}_dwi_mean.nii.gz -p centerline,${file_dwi}_dwi_mean_centerline.nii.gz -size 30mm
-# Motion correction
-sct_dmri_moco -i ${file_dwi}.nii.gz -bvec ${file_dwi}.bvec -m mask_${file_dwi}_dwi_mean.nii.gz -x spline
-file_dwi=${file_dwi}_moco
-file_dwi_mean=${file_dwi}_dwi_mean
+if [[ -e ${SUBJECT}_acq-b0_dwi.nii.gz ]]; then
+  echo "Found additional b=0 scans: $file_b0.nii.gz"
+  # Update global variable
+  FILE_DWI="${file_dwi}_concat"
+else
+  echo "No additional b=0 scans was found."
+  FILE_DWI="${file_dwi}"
+fi
+
+file_dwi_mean="${FILE_DWI}_moco_dwi_mean"
```

Comment:
we probably need to find a BIDS-compliant name for it... @alexfoias might have a clue

Reply:
I searched a bit, but I did find anything really relevant, right now, the name is for example:
`sub-amu01_dwi_concat_moco_dwi_mean.nii.gz` (I definitely would not keep that)
I found this --> https://bids-specification.readthedocs.io/en/stable/05-derivatives/02-common-data-types.html#preprocessed-or-cleaned-data
So we should add a json sidecar I think
Maybe the name could be : `sub-amu01_dwi_average.nii.gz`

Reply:
`sub-amu01_dwi_average.nii.gz` is definitely not compliant.
As for `sub-amu01_desc-average_dwi.nii.gz`, it is compliant, but `desc-` is an entity only allowed in the `derivatives` folder (see: https://bids-specification.readthedocs.io/en/stable/99-appendices/09-entities.html#desc)

So it depends on what you are doing here. Are those derivatives in an existing dataset or is this a new separate "preprocess" dataset?

We had a similar discussion thread about the vertebral labelling script [here](https://github.com/ivadomed/ivadomed/issues/707#issuecomment-829209716), where we decided to use the entity `rec-`. Maybe it could apply here.
`rec-` definition: https://bids-specification.readthedocs.io/en/stable/99-appendices/09-entities.html#rec

Reply:
> Are those derivatives in an existing dataset or is this a new separate "preprocess" dataset?

It's going to be a new preprocess dataset, so yes `rec-` would be good I think here! Thanks @mariehbourget 
So it would be something like : `sub-amu01_rec-average_dwi.nii.gz` ?
motion correction was also applied before averaging, we should add it?

Reply:
> motion correction was also applied before averaging, we should add it?

no need to elongate the file name further-- that information should be present in the README of the repository, which will/should include a link to the processing file (used to generate this processed dataset).

thank you @mariehbourget 🙏